test/agentchat/test_groupchat.py

```diff
@@ -671,6 +671,103 @@ def test_clear_agents_history():
         {"content": "How you doing?", "name": "sam", "role": "user"},
     ]
 
+    # testing saving tool_call message when clear history going to remove it leaving only tool_response message
+    agent1.reset()
+    agent2.reset()
+    agent3.reset()
+    # we want to broadcast the message only in the preparation.
+    groupchat = autogen.GroupChat(agents=[agent1, agent2, agent3], messages=[], max_round=1, enable_clear_history=True)
+    group_chat_manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=False)
+    # We want to trigger the broadcast of group chat manager, which requires `request_reply` to be set to True.
+    agent1.send("dummy message", group_chat_manager, request_reply=True)
+    agent1.send(
+        {
+            "content": None,
+            "role": "assistant",
+            "function_call": None,
+            "tool_calls": [
+                {"id": "call_test_id", "function": {"arguments": "", "name": "test_tool"}, "type": "function"}
+            ],
+        },
+        group_chat_manager,
+        request_reply=True,
+    )
+    agent1.send(
+        {
+            "role": "tool",
+            "tool_responses": [{"tool_call_id": "call_emulated", "role": "tool", "content": "example tool response"}],
+            "content": "example tool response",
+        },
+        group_chat_manager,
+        request_reply=True,
+    )
+    # increase max_round to 3
+    groupchat.max_round = 3
+    group_chat_manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=False)
+    with mock.patch.object(builtins, "input", lambda _: "clear history alice 1. How you doing?"):
+        agent1.initiate_chat(group_chat_manager, message="hello", clear_history=False)
+
+    agent1_history = list(agent1._oai_messages.values())[0]
+    assert agent1_history == [
+        {
+            "tool_calls": [
+                {"id": "call_test_id", "function": {"arguments": "", "name": "test_tool"}, "type": "function"},
+            ],
+            "content": None,
+            "role": "assistant",
+        },
+        {
+            "content": "example tool response",
+            "tool_responses": [{"tool_call_id": "call_emulated", "role": "tool", "content": "example tool response"}],
+            "role": "tool",
+        },
+    ]
+
+    # testing clear history called from tool response
+    agent1.reset()
+    agent2.reset()
+    agent3.reset()
+    agent2 = autogen.ConversableAgent(
+        "bob",
+        max_consecutive_auto_reply=10,
+        human_input_mode="NEVER",
+        llm_config=False,
+        default_auto_reply={
+            "role": "tool",
+            "tool_responses": [{"tool_call_id": "call_emulated", "role": "tool", "content": "example tool response"}],
+            "content": "Clear history. How you doing?",
+        },
```

Comment:
Is this a valid message? I thought that when `tool_responses` is not empty, the content should be empty.
@yenif is my assumption right?

Reply:
@sonichi tested it. When we printing something on tool response (making `content `non-empty), `tool_responses` is still here, but the internal `content `parameter is equal to "USER INTERRUPTED". Changed it in testes.